tls/client-ech.c

```diff
@@ -0,0 +1,194 @@
+/* client-ech.c
+ *
+ * Copyright (C) 2006-2022 wolfSSL Inc.
+ *
+ * This file is part of wolfSSL. (formerly known as CyaSSL)
+ *
+ * wolfSSL is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * wolfSSL is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA
+ */
+#include <wolfssl/options.h>
+#include <wolfssl/wolfcrypt/settings.h>
+#include <wolfssl/ssl.h>
+#include <wolfssl/test.h>
+#include <errno.h>
+
+#define SERV_PORT 443
+#define CERT_FILE "../certs/ech-client-cert.pem"
+
+#ifdef HAVE_ECH
+int main(void)
+{
+    int ret = 0;
+    byte rd_buf[512];
+    int sockfd = 0;
+    WOLFSSL_CTX* ctx = NULL;
+    WOLFSSL* ssl = NULL;
+    WOLFSSL_METHOD* method;
+    struct  sockaddr_in servAddr;
+    const char message[] =
+    "GET /cdn-cgi/trace/ HTTP/1.1\r\n"
+    "Host: crypto.cloudflare.com\r\n"
+    "User-Agent: Mozilla/5.0 (X11; Linux x86_64; rv:105.0) Gecko/20100101 Firefox/105.0\r\n"
+    "Accept: text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,*/*;q=0.8\r\n"
+    "Accept-Language: en-US,en;q=0.5\r\n"
+    "Referer: https://www.google.com/\r\n"
+    "DNT: 1\r\n"
+    "Connection: keep-alive\r\n"
+    "Upgrade-Insecure-Requests: 1\r\n"
+    "Sec-Fetch-Dest: document\r\n"
+    "Sec-Fetch-Mode: navigate\r\n"
+    "Sec-Fetch-Site: cross-site\r\n"
+    "Pragma: no-cache\r\n"
+    "Cache-Control: no-cache\r\n"
+    "\r\n";
+    const char ip_string[] = "162.159.137.85";
+    uint8_t ech_configs[72];
```

Comment:
Where does `72` come from?

Reply:
72 happens to be the length of the ech config that the cloudflare server uses but it can change so I should probably make it some larger generic number